src/algebra/lie/basic.lean

```diff
@@ -544,64 +549,139 @@ end lie_algebra
 
 section lie_module
 
-variables (R : Type u) (L : Type v) [comm_ring R] [lie_ring L] [lie_algebra R L]
-variables (M  : Type v) [add_comm_group M] [module R M]
+variables (R : Type u) {L : Type v} {M  : Type w}
+variables [comm_ring R] [lie_ring L] [lie_algebra R L] [add_comm_group M] [module R M]
 
 /-- A Lie module is a module over a commutative ring, together with a linear action of a Lie
 algebra on this module, such that the Lie bracket acts as the commutator of endomorphisms. -/
-class lie_module extends linear_action R L M :=
-(lie_act : ∀ (l l' : L) (m : M), act ⁅l, l'⁆ m = act l (act l' m) - act l' (act l m))
+class lie_module (α : has_bracket L M) :=
```

Comment:
Maybe worth noting that we could unbundle / break this down further as:
```lean
class lie_ring_module (L M : Type*) (α : has_bracket L M) [lie_ring L] [add_comm_group M] :=
(add_lie : ∀ (x y : L) (m : M), ⁅x + y, m⁆ = ⁅x, m⁆ + ⁅y, m⁆)
(lie_add : ∀ (x : L) (m n : M), ⁅x, m + n⁆ = ⁅x, m⁆ + ⁅x, n⁆)
(lie_lie : ∀ (x y : L) (m : M), ⁅⁅x, y⁆, m⁆ = ⁅x, ⁅y, m⁆⁆ - ⁅y, ⁅x, m⁆⁆)

class lie_algebra_module (R L M : Type*) (α : has_bracket L M)
  [comm_ring R] [lie_ring L] [lie_algebra R L] [add_comm_group M] [module R M] [lie_ring_module L M α] :=
(smul_lie : ∀ (t : R) (x : L) (m : M), ⁅t • x, m⁆ = t • ⁅x, m⁆)
(lie_smul : ∀ (t : R) (x : L) (m : M), ⁅x, t • m⁆ = t • ⁅x, m⁆)
```
to emphasise the analogy with the associative case:
```lean
class algebra_representation (R A M : Type*)
  [comm_ring R] [ring A] [algebra R A] [add_comm_group M] [module R M] [module A M]
  extends is_scalar_tower R A M
```

Reply:
Looking over this again now I think I might take another pass this weekend and see if I can carry out this further unbundling.